ARCHITECTURE.md

```diff
@@ -0,0 +1,226 @@
+# Istio operator architecture
+
+## Introduction
+
+The operator code is divided roughly into five areas:
+
+1. [New API](The_new_API) and related infrastructure. The new API is expressed as a
+[proto](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto) and
+compiled to [Go structs](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/istiocontrolplane_types.pb.go). The new
+API has pass-through fields to the Helm values.yaml API, but these are additionally validated through a
+[schema](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/values_types.go).
+1. [Controller](K8s_controller) code. The code comprises the K8s listener, webhook and logic for reconciling the cluster
+to an IstioControlPlane CR.
+1. [Manifest creation](Manifest_creation) code. Manifest rendering is done by overlaying user settings on top of a
```

Comment:
"Render Kubernetes manifests for the Istio Control Plane deployments from the user provided IstioControlPlane API specification. This involves applying the user's settings onto the selected base profile before Helm template rendering. Afterwards any optional user provided overlay patches are applied to the rendered manifests".

Reply:
I reworked this a bit to be clearer and avoid passive voice. lmkwyt